### DISCUSSION TITLE
Testing av komponenter i Jøkul

### DISCUSSION THREAD

**Discussion by piofinn** [fremtind/jokul#3118] [Annet]
# Kan vi teste Jøkul-komponentene bedre enn vi gjør i dag?

Som diskutert på forum kan det hende vi har noen hull i hvordan Jøkul-komponentene testes før de releases. For å få bedre oversikt over hvilke typer problemer som slipper gjennom, og hvordan (og om) vi kan teste for dem bedre, vil vi gjerne at dere bidrar! Derfor vil vi gjerne vite:

## Hvilke bugs har dere opplevd i produksjon med Jøkul-komponenter?

Vi trenger ikke nødvendigvis en utfyllende liste, men vi vil gjerne vite litt om **hva slags problemer** som dukker opp, og **i hvilke anledninger** problemene dukker opp. Er det for eksempel problemer med logikk som oppstår i forbindelse med skjemahåndtering, eller visuelle problemer som dukker opp på enkelte flater?

**Husk at dette er et åpent forum** slik at vi **IKKE KAN DELE INTERN KODE** i denne tråden. Det gjør det selvfølgelig litt verre å gi god kontekst for noen problemer, men dere får forklare så godt det lar seg gjøre uten å dele bedriftshemmeligheter.

**Comment by SercanSercan** [comment]
Takk for denne conversation, @piofinn. Flest problemer som vi har opplevde i SeOpp er relatert til cross-browser eller integration testing. 

**Cross-browser testing**
Tester alle utviklere sin implementasjon på flere nettlesere før en component blir prodsatt? 
Unit testing hjelper ikke å oppdage bug i ulike nettlesere. Jeg vet ikke om det finnes "automated cross-browser testing" tools. Bør vi legge det til i våre utvikling prosess? Ellers bør vi stole på mennesker(utvikler) med nye retningslinjer som er mer synlig? (Se eksempel her: https://ewig5qf9cgn.exactdn.com/wp-content/uploads/2020/08/Screen-Shot-2020-08-06-at-10.52.45-PM.jpg?strip=all&lossy=1&ssl=1)

**Integration testing**
Vi (utviklere) bruker portalen til å teste/debugge imens vi jobber på en sak, men portalen hjelper deg ikke å se dine endringer i en ekte web applikasjon. (Det er en bug for eksempel i Cookie-consent som skroller til helt bunnen av side) Derfor vi håper at våre endringer skal fungere fint i ulike applikasjoner. Skulle vi ha en tester stilling i kjerne-team som skal bekrefte løsninger for oss? (wild-card tester :) )

Ellers, vi gjør ingen endring i Jøkul's utvikling/prodsetting prosess, men ber alle team til å teste sin egen produkt bedre når det gjelder endringer i Jøkul. Beklager at min kommentar er ganske rotete, men jeg tenker at vi bør velge en vei som funker mest effektiv for alle.

**Reply by piofinn** [reply]
> Vi (utviklere) bruker portalen til å teste/debugge imens vi jobber på en sak, men portalen hjelper deg ikke å se dine endringer i en ekte web applikasjon.

Har dere sett på denne guiden for å teste Jøkul-pakkene i egen applikasjon? Tenker at det vil kunne forbedre akkurat dette punktet en del for dere :) https://jokul.fremtind.no/guider/hvordan-teste-endringer-eget-prosjekt

**Reply by SercanSercan** [reply]
Takk, Pio :) Jeg skal grave den.
